tmt/steps/prepare/brand_new_allmighty_install/providers/brew.py

```diff
@@ -0,0 +1,15 @@
+"""
+Brew Artifact Provider
+"""
+
+import tmt.log
+from tmt.steps.prepare.brand_new_allmighty_install.providers.koji import KojiProvider
+
+
+class BrewProvider(KojiProvider):
+    """
+    Brew builds are just a special case of Koji builds
+    """
+
+    def __init__(self, logger: tmt.log.Logger, artifact_id: str):
+        super().__init__(logger, artifact_id)
```

Comment:
The specific `__init__` should not be necessary, it should automatically inherit the parent's. For dataclasses it definitely makes it cleaner and should not be added, but for normal classes it can be either way. I see there is quite a lot of precedence of keeping these, some of which have a reasoning, but I am not sure about the vast majority of those usages.

Reply:
We add explicit `__init__` when inheriting from `tmt.utils.Common`, because inheritance is tricky there, and without calling `super().__init__()`, it does not work correctly. Here we can drop it if it does not add anything new, and I'd guess that `self._logger = logger` and `self.id = artifact_id` would be somehow common to all classes, therefore I'd expect that to happen in `ArtifactProvider.__init__()` already.